app-editors/atom/atom-1.7.3.ebuild

```diff
@@ -2,84 +2,361 @@
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=6
+EAPI=5
 
 PYTHON_COMPAT=( python2_7 )
-inherit flag-o-matic python-any-r1 eutils
+inherit flag-o-matic python-any-r1 eutils rpm
 
 DESCRIPTION="A hackable text editor for the 21st Century"
 HOMEPAGE="https://atom.io"
-SRC_URI="https://github.com/atom/atom/archive/v${PV}.tar.gz -> ${PV}.tar.gz"
+MY_PV="${PV//_/-}"
+
+# All binary packages depend on this
+NAN_V=2.0.9
+
+# Keep dep versions in sync with atom/package.json
+CACHED_RUN_IN_THIS_CONTEXT_V=0.4.1
+GIT_UTILS_V=4.1.2
+NODEGIT_V=0.12.2
+NODE_NSLOG_V=3.0.0
+NODE_ONIGURUMA_V=5.1.2
+NODE_PATHWATCHER_V=6.2.4
+NODE_RUNAS_V=3.1.1
+SCROLLBAR_STYLE_V=3.2.0
+SPELL_CHECK_V=0.67.0
+
+# textbuffer dependency
+MARKER_INDEX_V=3.1.0
+
+# apm dependency
+NODE_KEYTAR_V=3.0.2
+
+# atom-keymap dependency
+KEYBOARD_LAYOUT_V=1.0.0
+
+# symbols-view dependency
+NODE_CTAGS_V=3.0.0
+
+# spell-check dependency
+NODE_SPELLCHECKER_V=3.2.3
+
+# nodegit dependencies
+PROMISIFY_NODE_V=0.4.0
+NODE_FS_EXTRA_V=0.26.2
+NODE_GRACEFUL_FS_V=4.1.2
+NODE_JSONFILE_V=2.1.0
+NODE_KLAW_V=1.0.0
+RIMRAF_V=2.2.8
+NODE_LODASH_V=3.10.1
+COMBYNE_V=0.8.1
+JS_BEAUTIFY_V=1.5.10
+NODEGIT_PROMISE_V=4.0.0
+ASAP_V=2.0.3
+OBJECT_ASSIGN_V=4.0.1
+
+# The x86_64 arch below is irrelevant, as we will rebuild all binary packages.
+SRC_URI="
+	https://github.com/${PN}/${PN}/releases/download/v${MY_PV}/atom.x86_64.rpm -> atom-${MY_PV}.rpm
```

Comment:
Why do you use .rpm instead of .tar.gz tarball when the latter is available?
Current ebuild in the tree uses .tar.gz tarball.